CIP-???? | Governance Participant Compensation

Thomas-nada opened the pull request:
This  proposal introduces Governance Participant Compensation to provide automated, deterministic compensation for Cardano governance actors (CC, DReps, SPOs) who cast a valid vote.

It achieves this by:

Funding: Consuming a portion of the proposal's deposit (not the Treasury) upon finalization.

Distribution: Using new ledger rules to split the funds into pools and distributing them via a hybrid model (equal share + stake-weighted) only to active voters.

Mechanism: Operating entirely within the ledger rules (like staking rewards) to ensure reliability and trustlessness.

Goal: Ensure sustainable, decentralized governance by making participation economically feasible.

---

([rendered latest proposal](https://github.com/Thomas-nada/CIPs/blob/Compensatory-Ledger-Mechanism/CIP-%3F%3F%3F%3F/README.md))

Thomas-nada committed:
Create README.md

Thomas-nada committed:
Update README.md

Thomas-nada committed:
Added discussion

rphair reviewed (commented):
Marked for `Triage` in next CIP meeting https://hackmd.io/@cip-editors/124.

Thomas-nada committed:
Update CIP-????/README.md

Co-authored-by: Robert Phair <rphair@cosd.com>

Thomas-nada committed:
Update CIP-????/README.md

Co-authored-by: Robert Phair <rphair@cosd.com>

Thomas-nada committed:
Update CIP-????/README.md

Co-authored-by: Robert Phair <rphair@cosd.com>

Thomas-nada committed:
Update CIP-????/README.md

Co-authored-by: Robert Phair <rphair@cosd.com>

Thomas-nada committed:
Update README.md

Thomas-nada commented on CIP-????/README.md:
```diff
@@ -0,0 +1,244 @@
+
+Cardano’s decentralized governance framework requires persistent engagement from CC members, DReps, and SPOs. These actors must analyze governance actions, evaluate technical and economic implications, and cast informed votes. This work incurs ongoing time, expertise, and operational costs. Without a reliable compensation model, several risks emerge:
+
+### **1. Governance Fatigue**
```
Updated

Thomas-nada committed:
Update README.md

Thomas-nada committed:
Update README.md

Thomas-nada committed:
Create example.md

Thomas-nada committed:
Add reward-account extensions and eligibility rules

Updated to add optional reward-account support for DReps and CC hot keys, plus the eligibility rules for who can actually receive CLM payouts.

This includes:
- new ledger maps for storing optional reward accounts
- updated DRep registration cert
- updated CC hot key cert
- eligibility section (must vote + must have reward account)

Thomas-nada committed:
Minor wording fix

Thomas-nada committed:
Updated the CIP so that only governance groups **relevant** to a given action type generate compensation. Added the `EffectiveRate` logic, updated the compensation pot formulas, and fixed the deposit/refund rules so that any portion tied to non-relevant groups isn’t consumed and is returned to the proposer.

This makes the CLM behave correctly when, for example, SPOs can vote but their votes aren’t counted towards ratification. Only DReps/CC get compensated, and the SPO portion of the deposit stays refundable.

Ryun1 commented on CIP-????/README.md:
```diff
@@ -0,0 +1,311 @@
+Cardano’s decentralized governance framework requires persistent engagement from CC members, DReps, and SPOs. These actors must analyze governance actions, evaluate technical and economic implications, and cast informed votes. This work incurs ongoing time, expertise, and operational costs. Without a reliable compensation model, several risks emerge:
+
+### Governance Fatigue
+Uncompensated governance labor leads to declining participation over time.
```
have we seen this?
do we have numbers on this?

Ryun1 commented on CIP-????/README.md:
```diff
@@ -0,0 +1,311 @@
+---
+CIP: XXXX
+Title: Compensatory Ledger Mechanism
```
```suggestion
Title: Governance Participant Compensation
```

I feel something referring to Governance Participants is probably more accurate and obvious at first glance

rphair replied:
yes... my first reading of the title also (before I got around to reading the document) was that this was some accounting capability and using "compensatory" in the mathematical sense (rather than "compensation").

Ryun1 commented on CIP-????/README.md:
```diff
@@ -0,0 +1,311 @@
+
+**Constraint:**  
+`ccCompensationRate + drepCompensationRate + spoCompensationRate ≤ MAX_COMP_RATE`  
+where `MAX_COMP_RATE` is hard-coded (e.g., 0.25).
```
probably not good to hard-code
best to have everything as an adjustable protocol parameter

fallen-icarus commented:
The idea I brought up in the CIP meeting was to use [CIP-159](https://github.com/cardano-foundation/CIPs/pull/1061) for micro-taxes:

_When a user delegates to a dRep, their wallet automatically subscribes them into paying 0.001 ADA to their dRep every time they make a tx._

The user can always opt-out, but opt-out still typically sees >80% participation rates. We can do the same sort of micro-taxes for funding other entities like the CC or stake pool operators running the testnets.

Thomas-nada commented on CIP-????/README.md:
```diff
@@ -0,0 +1,311 @@
+Without compensation, new and smaller actors lack incentives to engage, harming governance quality.
+
+### Misaligned Incentives
+Proposers incur no cost commensurate with the burden placed on reviewers and voters.
```
True, but is that considered cost? They are not paying for it. They incur potential losses, not cost.

Ryun1 replied:
I feel probably best to mention this

Thomas-nada committed:
Update CIP-????/README.md

Co-authored-by: Ryan <ryan.williams@intersectmbo.org>

Thomas-nada commented on CIP-????/README.md:
```diff
@@ -0,0 +1,311 @@
+Proposers incur no cost commensurate with the burden placed on reviewers and voters.
+
+### Need for a Deterministic, Automated Mechanism
+Any off-chain or trust-based compensation scheme introduces friction and reduces security.
```
Agreed. If a smart contract solution can do the same with less changes to the ledger I am all for it.

Thomas-nada committed:
Update CIP-????/README.md

Co-authored-by: Ryan <ryan.williams@intersectmbo.org>

Thomas-nada commented:
> The idea I brought up in the CIP meeting was to use [CIP-159](https://github.com/cardano-foundation/CIPs/pull/1061) for micro-taxes:
> 
> _When a user delegates to a dRep, their wallet automatically subscribes them into paying 0.001 ADA to their dRep every time they make a tx._
> 
> The user can always opt-out, but opt-out still typically sees >80% participation rates. We can do the same sort of micro-taxes for funding other entities like the CC or stake pool operators running the testnets.

With [CIP-149 ](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0149) currently being rolled out by wallet providers, SDKs and explorers we will be able to see the adpotion rate of a similar approach and also how the distribution rate will be. 
If this is a success I am happy to go this route. But I am for the most against relying on voluntarism as the backbone of compensation. I draw similarities with waiters who are poorly paid because their employers know the waiters can rely on tips.

Cerkoryn commented on CIP-????/README.md:
```diff
@@ -0,0 +1,311 @@
+
+---
+
+#### DRep Compensation (Hybrid Model)
```
@Ryun1 are you by chance referring to the Threshold(k) mechanism proposed in this paper?

https://iohk.io/en/research/library/papers/reward-schemes-and-committee-sizes-in-proof-of-stake-governance

rphair reviewed (commented):
@Thomas-nada as it says in CIP-0001: submissions to this repository often require an _advocate_ (usually the author) who posts regularly to keep bringing editorial & peer review together with community statements & ideas and providing responses to feedback... with requests for re-review every time consensus needs to be, or appears to be, achieved.

So far this thread has been like a posting board for ideas on the subject, without any emphatic confirmation that the proposed method would be acceptable to anyone.  But perhaps now that this has been merged (cc @maureenblack @daniellestanko-gov - with maybe @thenic95 also interested):
* [CPS-0033 | DRep Voting Power Concentration](https://github.com/cardano-foundation/CIPs/tree/master/CPS-0033)

... we might clarify these discussions, and you might offer this proposal back to the relevant communities — to see if they can provide some evidence that they would find this proposal acceptable — and to implement some changes if necessary to gain their support.
* Of course you would be welcome to use this thread to hold such discussions: as the CIP Editors generally hope for in the review process.
* If we again see some progress beyond the above inconclusive review threads, we can get this proposal back in the CIP Meeting review agenda for hopeful confirmation as a CIP candidate (with a number) & then consideration as a real technical proposition of interest to the governance community.

Conversely: without that progress, this proposal would proceed toward closure: not because it isn't a good idea, but because we can't use it as a CIP without the vital component of conclusive peer review.  cc @Ryun1 @perturbing + reviewers @Cerkoryn @fallen-icarus
